src/Microsoft.AspNet.Razor/Generator/Compiler/CodeBuilder/CSharp/CSharpTagHelperCodeRenderer.cs

```diff
@@ -69,10 +69,9 @@ public void RenderTagHelper(TagHelperChunk chunk)
 
             // Determine what attributes exist in the element and divide them up.
             var htmlAttributes = chunk.Attributes;
-            var attributeDescriptors = tagHelperDescriptors.SelectMany(descriptor => descriptor.Attributes);
-            var unboundHtmlAttributes = htmlAttributes.Where(
-                attribute => !attributeDescriptors.Any(
-                    descriptor => string.Equals(attribute.Key, descriptor.Name, StringComparison.OrdinalIgnoreCase)));
+            var attributeDescriptors = tagHelperDescriptors.SelectMany(descriptor => descriptor.Attributes).ToArray();
```

Comment:
Because the enumeration is executed as many times as there are HTML attributes in the line immediately below.